content/tutorials/history.textile

```diff
@@ -375,7 +453,25 @@ In this demo we use the "history API":/realtime/history to replay up to the last
 h2. Download tutorial source code
 
 blang[java].
-  The complete source code for each step of "this tutorial is available on Github":https://github.com/piotrekinventica/tutorials/commits/history-java-rest.
+  The complete source code for each step of "this tutorial is available on Github":https://github.com/piotrekinventica/tutorials/commits/history-jre-rest.
+
+  We recommend that you clone the repo locally:
+
+  bc[sh]. git clone https://github.com/ably/tutorials.git
+
+  Checkout the tutorial branch:
+
+  bc[sh]. git checkout history-jre-rest
+
+  Then you can run project inside your console. Be sure to switch into project's directory and then use these commands in your terminal:
+
+  bc[sh]. ./gradlew assemble
+  ./gradlew run  
+
+  Don't forget to replace your @ExampleActivity#API_KEY@ field with "Ably API key":https://support.ably.io/support/solutions/articles/3000030502-setting-up-and-managing-api-keys.
+
+blang[android].
+  The complete source code for each step of "this tutorial is available on Github":https://github.com/piotrekinventica/tutorials/commits/history-java.
```

Comment:
wrong repo, and commits should be `history-android` if you are doing `history-jre-rest` for consistency, not `history-java`